tests/users.spec.ts

```diff
@@ -0,0 +1,118 @@
+import {
+    describe, test, expect
+} from '@jest/globals';
+import {UsersService} from "./services/userService.ts";
+import {generateUserData} from "./helpers/user-data-generator.ts";
+import {IUser} from "./types.ts";
+import {UserRepository} from "./repositories/userRepository.ts";
+import {DBClient} from "./db-client/DBClient.ts";
+import {Client} from "ts-postgres";
+import {uuid} from "../src/utils.ts";
+
+const usersService = new UsersService();
+let client: Client;
+let userRepository: UserRepository;
+
+beforeAll(async () => {
+    client = await DBClient.createAndConnectClient();
+    userRepository = new UserRepository(client);
+});
+
+afterAll(async () => {
+    await client.end();
+});
+
+
+describe('Users endpoints', () => {
+    test('/users POST Should create a new user', async () => {
+        const userData = generateUserData();
+        const newUserResponse = await usersService.createUser(userData);
+        const createdUserData = newUserResponse.data.data;
+
+        expect(newUserResponse.status).toBe(200);
+        expect(createdUserData.id).toBeDefined();
+        expect(createdUserData.username).toEqual(userData.username);
+
+        const createdUserFromDB = await userRepository.selectOne(createdUserData.id);
+
+        if (createdUserFromDB !== null) {
+            //database types dictate this decision
+            expect(createdUserData.id).toEqual(createdUserFromDB.id);
+            expect(userData.username).toEqual(createdUserFromDB.username);
+            expect(userData.roleId).toEqual(createdUserFromDB.roleId);
+            expect(userData.firstName).toEqual(createdUserFromDB.firstName);
+            expect(userData.lastName).toEqual(createdUserFromDB.lastName);
+            expect(userData.phoneNumber).toEqual(createdUserFromDB.phoneNumber);
+            expect(userData.email).toEqual(createdUserFromDB.email);
+            expect(userData.telegram).toEqual(createdUserFromDB.telegram);
```

Comment:
Its better to have expect(expectedUser).toDeepEqual(actualUser) or something like that. In case our user is extended, we will need to go through all such code. Its better to incapsulate such logic.

Reply:
Totally agree, but I faced an issue with types returned from the DB.